lib/pure/collections/deques.nim

```diff
@@ -472,7 +459,22 @@ proc `$`*[T](deq: Deque[T]): string =
     result.addQuoted(x)
   result.add("]")
 
-
+proc hash*[A](d: Deque[A]): Hash =
+  ## Hashing of Deque.
+  runnableExamples:
+    var
+      x: Deque[int]
+      y: Deque[int]
+
+    for i in 1..5:
+      x.addLast(i*10)
+    for i in -5..10:
+      y.addLast(i*10)
+    y.shrink(fromFirst = 6, fromLast = 5)
+    assert hash(x) == hash(y)
+  for h in 0 ..< d.len:
+    result = result xor hash(d[h])
```

Comment:
this is a terrible hash!
if d is composed of consecutive pairs, eg `[x1, x1, x2, x2, x3, x3]`, you'll get 0...
instead use the standard `!&` for mixing see https://nim-lang.github.io/Nim/hashes.html
eg: `result = result !& hash(d[h])`

Reply:
Right, not sure where I got that algorithm from TBH. Thought I copied it from some other module but I'm not able to find which..

Reply:
Ah, appears like it was the `sets` module:
```nim
proc hash*[A](s: HashSet[A]): Hash =                                                                                                                                                
  ## Hashing of HashSet.                                                                                                                                                            
  for h in 0 .. high(s.data):                                                   
    result = result xor s.data[h].hcode                                         
  result = !$result
```

Reply:
Ugh, we shd fix that too!